front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -211,10 +217,42 @@ const SpaceTimeChartWrapper = ({
     allTrainsProjected
   );
 
-  const splitPoints = useMemo<SplitPoint[]>(() => {
-    const pathsIndex = keyBy(paths, ({ id }) => id);
+  const pathsById = useMemo(() => keyBy(paths, ({ id }) => id), [paths]);
+
+  const toPaced = (id: string) => {
+    if (isOccurrenceId(id)) return extractPacedTrainIdFromOccurrenceId(id);
+    if (isPacedTrainId(id)) return id;
+    return undefined;
+  };
+
+  const countZonesByPacedTrainId = (zones: OccupancyZone[] = []) => {
+    const counts = new Map<TimetableItemId, Map<string, number>>();
+    for (const zone of zones) {
+      if (isOccurrenceId(zone.trainId) && zone.trackId) {
```

Comment:
I think that `trackId` is always defined (according to Typescript)
```suggestion
      if (isOccurrenceId(zone.trainId)) {
```